sound/soc/sof/topology.c

```diff
@@ -1993,11 +1995,23 @@ static int sof_link_ssp_load(struct snd_soc_component *scomp, int index,
 		return ret;
 	}
 
-	/* set config for all DAI's with name matching the link name */
-	ret = sof_set_dai_config(sdev, size, link, config);
-	if (ret < 0)
-		dev_err(sdev->dev, "error: failed to save DAI config for SSP%d\n",
-			config->dai_index);
+	/* set config to dai component of playback */
+	if (link->dpcm_playback) {
+		ret = sof_set_dai_config(sdev, size, link, config,
+					 SNDRV_PCM_STREAM_PLAYBACK);
+		if (ret < 0)
+			dev_err(sdev->dev, "error: failed to save DAI config in dai component of playback for SSP%d\n",
+				config->dai_index);
+	}
+
+	/* set config to dai component of capture */
+	if (link->dpcm_capture) {
+		ret = sof_set_dai_config(sdev, size, link, config,
+					 SNDRV_PCM_STREAM_CAPTURE);
+		if (ret < 0)
+			dev_err(sdev->dev, "error: failed to save DAI config in dai component of capture for SSP%d\n",
+				config->dai_index);
+	}
```

Comment:
question: is there a way to avoid code copy/paste with just s/playback/capture changed, maybe with a loop or something smarter?

Reply:
yes, I will try